extensions/schema-compare/src/models/schemaCompareOptionsModel.ts

```diff
@@ -4,222 +4,74 @@
  *--------------------------------------------------------------------------------------------*/
 import * as loc from '../localizedConstants';
 import * as mssql from 'mssql';
+import * as vscode from 'vscode';
 import { isNullOrUndefined } from 'util';
 
 export class SchemaCompareOptionsModel {
-	public deploymentOptions: mssql.DeploymentOptions;
-	public excludedObjectTypes: mssql.SchemaObjectType[] = [];
-	public optionsMapTable: Map<string, mssql.DacDeployOptionPropertyBoolean>;
-
-	public optionsLookup = {};
+	// key is the option display name and values are checkboxValue and optionName
+	private optionsValueNameLookup: { [key: string]: mssql.IOptionWithValue } = {};
+	public excludedObjectTypes: number[] = [];
 	public objectsLookup = {};
 
-	//#region Schema Compare Deployment Options
+	constructor(public deploymentOptions: mssql.DeploymentOptions) {
```

Comment:
As with the other - I would suggest having this be a property setter instead that then calls `setOptionsToValueNameLookup` - since you'll always want to ensure the lookup map is reset when the deploymentOptions change.